_posts/news/2021-08-04-announcing-scalajs-1.7.0.md

```diff
@@ -0,0 +1,135 @@
+During the implementation of this feature, we have analyzed the corpus of all Scala.js-only libraries (without JVM support) and extracted all the regexes that they use.
+We have verified that none of those use cases are impacted by this change.
+It is still possible that applications are impacted.
+
+It is also possible for some cross-platform libraries to face issues, as we have not covered those.
+Unlike Scala.js-only libraries, we consider it unlikely that they will have issues with the *change of semantics* per se, as they already worked on the JVM, with the new semantics.
+
+The biggest danger would be a cross-library that uses the `MULTILINE` flag (aka `(?m)`).
+Indeed, that feature kind of worked before, but is now rejected at `Pattern.compile()`-time with a `PatternSyntaxException` by default.
+The reason is that to correctly implement the semantics of that flag, we need support for look-behind assertions (`(?<=𝑋)`) in JavaScript's `RegExp`.
+That support was only added in ECMAScript 2018, whereas Scala.js targets ES 2015 by default.
+
+It is possible to change that target with the following setting:
+
+{% highlight scala %}
+scalaJSLinkerConfig ~= (_.withESFeatures(_.withESVersion(ESVersion.ES2018)))
+{% endhighlight %}
+
+**Attention!** While this enables support for the `MULTILINE` flag (among others), it restricts your application to environments that support recent JavaScript features.
+If you maintain a library, this restriction applies to all downstream libraries and applications.
+
+We therefore recommend to try and *avoid* the need for that flag instead.
+We give several strategies on how to do so [on the Regular expressions documentation page]({{ BASE_PATH }}/doc/regular-expressions.html).
+That page also contains many more details on the new support of regular expressions.
+
+## New features
+
+### Add a configurable header comment in generated .js files
+
+Sometimes, it is desirable to add a header comment in the generated .js files.
+This is typically used for license information or any other metadata.
+While it has always been possible to post-process the generated .js files in the build, doing so came at the cost of destroying the source maps.
+
+Scala.js 1.7.0 introduces a new linker configuration, `jsHeader`, to specify a comment to insert at the top of .js files:
+
+{% highlight scala %}
+scalaJSLinkerConfig ~= {
+  _.withJSHeader(
+    """
+      |/* This is the header, which source maps
+      | * take into account.
+      | */
+    """.stripMargin.trim() + "\n"
+  )
+},
+{% endhighlight %}
+
+The `jsHeader` must be a combination of valid JavaScript whitespace and/or comments, and must not contain any newline character other than `\n` (the UNIX newline).
+If non-empty, it must end with a new line.
+These restrictions ensure that this feature is not abused to inject arbitrary JavaScript code in the .js file generated by the compiler, potentially compromising the compiler abstractions.
+
+## Miscellaneous
```

Comment:
Should we mention that we removed the implementation restriction of default parameters for delegating JS class constructors?